Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -184,34 +318,56 @@ void HiresTexture::Prefetch()
       }
       if (iter != s_textureCache.end())
       {
-        for (const Level& l : iter->second->m_levels)
+        for (const Level& tex : iter->second->m_levels)
         {
-          size_sum += l.data_size;
+          prefetched_size_sum += tex.data_size;
         }
       }
     }
 
-    if (s_textureCacheAbortLoading.IsSet())
+    if (s_texturePrefetcher_AbortLoading.IsSet())
     {
       return;
     }
 
-    if (size_sum > max_mem)
+    // The important part - Stops prefetching to avoid getting out of RAM completely, crash prone
+    // Currently no way to know how much it's used by OS and other applications
+    if (prefetched_size_sum >= delegated_max_mem)
     {
       Config::SetCurrent(Config::GFX_HIRES_TEXTURES, false);
 
+      WARN_LOG(VIDEO, "Prefetcher: Max delegated mem breached, aborting further prefetching");
       OSD::AddMessage(
-          StringFromFormat(
-              "Custom Textures prefetching after %.1f MB aborted, not enough RAM available",
-              size_sum / (1024.0 * 1024.0)),
-          10000);
+          StringFromFormat("Prefetcher: Warning, custom textures prefetching after %.2f MB aborted."
+                           " Maximum delegated memory limit of %.2f breached.",
+                           prefetched_size_sum / (1024.0 * 1024.0),
+                           delegated_max_mem / (1024.0 * 1024.0)),
+          20000);
       return;
     }
-  }
-  u32 stoptime = Common::Timer::GetTimeMs();
-  OSD::AddMessage(StringFromFormat("Custom Textures loaded, %.1f MB in %.1f s",
-                                   size_sum / (1024.0 * 1024.0), (stoptime - starttime) / 1000.0),
-                  10000);
+
+  }  // End Prefetching
+  prefetch_stoptime_total = Common::Timer::GetTimeMs();
+
+  FileType_Counter_Total_Amount = FileType_Counter_DDS_Amount + FileType_Counter_PNG_Amount +
+                                  FileType_Counter_BMP_Amount + FileType_Counter_TGA_Amount +
+                                  FileType_Counter_JPG_Amount + FileType_Counter_HDR_Amount;
+  const std::string FT_C_DDS_s = std::to_string(FileType_Counter_DDS_Amount);
+  const std::string FT_C_PNG_s = std::to_string(FileType_Counter_PNG_Amount);
+  const std::string FT_C_BMP_s = std::to_string(FileType_Counter_BMP_Amount);
+  const std::string FT_C_TGA_s = std::to_string(FileType_Counter_TGA_Amount);
+  const std::string FT_C_JPG_s = std::to_string(FileType_Counter_JPG_Amount);
+  const std::string FT_C_HDR_s = std::to_string(FileType_Counter_HDR_Amount);
+  const std::string FT_C_Total_s = std::to_string(FileType_Counter_Total_Amount);
+  const double Pref_sum_total_MB = prefetched_size_sum / (1024.0 * 1024.0);
+  const double Pref_total_time_Sec = (prefetch_stoptime_total - prefetch_starttime_total) / 1000.0;
+
+  OSD::AddMessage("Prefetcher: Total of " + FT_C_Total_s + " custom textures loaded, " +
```

Comment:
Unfortunately this is a major feature of this PR  for me. Without this the PR is worthless. The whole point are more OSD messages for details that I alwa

Remember, as the -dev channel people said, some of them don't really care about custom textures and other things what I suggested, but they don't mind if I go and do things myself, so if everyone else puts in what they want, what can't I.

Since when is opensource some subjectivelly controlled thing ... I'm not paid to babysit the community.